### PR TITLE
Update Go minimum version

### PR DESCRIPTION
**Pull request by SuperQ**
* Set Go minimum version to 1.17.
* Update go.mod format for 1.17.
* Remove unecessary exclude block for k8s.io/client-go.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Review by Nexucis — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -100,32 +186,6 @@ exclude (
...
-	k8s.io/client-go v5.0.0+incompatible
-	k8s.io/client-go v5.0.1+incompatible
-	k8s.io/client-go v6.0.0+incompatible
-	k8s.io/client-go v7.0.0+incompatible
-	k8s.io/client-go v8.0.0+incompatible
-	k8s.io/client-go v9.0.0+incompatible
-	k8s.io/client-go v9.0.0-invalid+incompatible
-)
-
 retract (
```
Comment (roidelapluie) [inline go.mod:189]: This block can be removed as well

**Review by metalmatze — APPROVED** [review]
Nice. Makes sense to me with the context I have around the Kubernetes block. :+1:

**Comment by roidelapluie** [comment]
> Nice. Makes sense to me with the context I have around the Kubernetes block. +1

The kubernetes block can be removed because those versions have now been retracted upstream

File: go.mod (hunk 1 above)
Comment (metalmatze) [reply go.mod:189]: :broom:

Comment (SuperQ) [reply go.mod]: Done.

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**